packages/webdriverio/src/middlewares.js

```diff
@@ -21,31 +22,35 @@ export const elementErrorHandler = (fn) => (commandName, commandFn) => {
                 `that wasn't found, waiting for it...`
             )
 
-            return fn(commandName, () => {
-                /**
-                 * create new promise so we can apply a custom error message in cases waitForExist fails
-                 */
-                return new Promise((resolve, reject) => this.waitForExist().then(resolve, reject)).then(
-                    /**
-                     * if waitForExist was successful requery element and assign elementId to the scope
-                     */
-                    () => {
-                        return this.parent.$(this.selector).then((elem) => {
-                            this.elementId = elem.elementId
-                            return fn(commandName, commandFn).apply(this, args)
-                        })
-                    },
-                    /**
-                     * if waitForExist failes throw custom error
-                     */
-                    () => {
-                        throw new Error(`Can't call ${commandName} on element with selector "${this.selector}" because element wasn't found`)
-                    }
-                )
-            }).apply(this)
+            /**
+             * create new promise so we can apply a custom error message in cases waitForExist fails
+             */
+            try {
+                await this.waitForExist()
+            } catch (error) {
```

Comment:
`(error)` is redundant if you don't use it

Reply:
Its not redundant when its necessary.  You HAVE to catch with a variable, even when you don't intend to use it.

Reply:
You have to catch an error with a variable even if you don't intend to use it.

Reply:
I would say, let's add https://babeljs.io/docs/en/babel-plugin-proposal-optional-catch-binding and avoid the variable

Reply:
Done.